config/kube.py

```diff
@@ -0,0 +1,337 @@
+                ignore_error=True)
+
+    clicommon.run_command("kubeadm reset -f", ignore_error=True)
+    clicommon.run_command("rm -rf /etc/cni/net.d")
+    clicommon.run_command("rm -f {}".format(KUBE_ADMIN_CONF))
+    clicommon.run_command("systemctl stop kubelet")
+    clicommon.run_command("systemctl disable kubelet")
+
+
+def kube_join(force=False):
+    lock_fd = _take_lock()
+    if not lock_fd:
+        log.log_error("Lock {} is active; Bail out".format(LOCK_FILE))
+        return
+
+    db_data = Db().get_data('KUBERNETES_MASTER', 'SERVER')
+    if not db_data or 'IP' not in db_data or not db_data['IP']:
+        log.log_error("Kubernetes server is not configured")
+
+    if db_data['disable'].lower() != "false":
+        log.log_error("kube join skipped as kubernetes server is marked disabled")
+        return
+
+    if not force:
+        if _is_connected(db_data['IP']):
+            # Already connected. No-Op
+            return
+
+    kube_reset()
+    _do_join(db_data['IP'], db_data['insecure'])
+
+
+@click.group(cls=clicommon.AbbreviationGroup)
+def kubernetes():
+    """kubernetes command line"""
+    pass
+
+
+# cmd kubernetes join [-f/--force]
+@click.command()
+@click.option('-f', '--force', help='Force a join', is_flag=True)
+def join(force):
+    kube_join(force=force)
+
+kubernetes.add_command(join)
+
+
+# cmd kubernetes reset
+@click.command()
+def reset():
+    kube_reset()
+
+kubernetes.add_command(reset)
+
+
+# cmd kubernetes server
+@click.group()
+def server():
+    """ Server configuration """
+    pass
+
+kubernetes.add_command(server)
+
+
+# cmd kubernetes server IP
+@click.command()
+@click.argument('vip')
+def ip(vip):
+    """Specify a kubernetes cluster VIP"""
+    if not netaddr.IPAddress(vip):
+        click.echo('Invalid IP address %s' % vip)
+        return
+    _update_kube_server('IP', vip)
+
+server.add_command(ip)
+
+
+# cmd kubernetes server insecure
+@click.command()
+@click.argument('option', type=click.Choice(["on", "off"]))
+def insecure(option):
+    """Specify a kubernetes cluster VIP access as insecure or not"""
+    _update_kube_server('insecure', option == "on")
+
+server.add_command(insecure)
+
+
+# cmd kubernetes server disable
+@click.command()
+@click.argument('option', type=click.Choice(["on", "off"]))
+def disable(option):
+    """Specify a kubernetes cluster VIP access is disabled or not"""
+    _update_kube_server('disable', option == "on")
+
+server.add_command(disable)
+
+
+# cmd kubernetes label
+@click.group()
+def label():
+    """ label configuration """
+    pass
+
+kubernetes.add_command(label)
+
+
+# cmd kubernetes label add <key> <val>
+@click.command()
+@click.argument('key', required=True)
+@click.argument('val', required=True)
+def add(key, val):
+    """Add a label to this node"""
+    if not key or not val:
+        click.echo('Require key & val')
+        return
+    _label_node("{}={}".format(key, val))
+
+label.add_command(add)
```

Comment:
Rather than calling `kubernetes.add_command(x)`, you should simply be able to change the decorator from:

```python
@click.command()
```

to:

```python
@label.command()
```

Reply:
done